pycalphad/model.py

```diff
@@ -301,6 +300,13 @@ def degree_of_ordering(self):
     mixing_entropy = SM_MIX = property(lambda self: -self.GM_MIX.diff(v.T))
     mixing_heat_capacity = CPM_MIX = property(lambda self: -v.T*self.GM_MIX.diff(v.T, v.T))
 
+    @property
+    def reference_model(self):
+        """Reference model built as needed improve performance."""
+        if self._reference_model is None:
+            self.build_reference_model(self._dbe)
+        return self._reference_model
+
     def build_reference_model(self, dbe, preserve_ideal=True):
```

Comment:
`build_reference_model` is mixing user-provided state (`dbe`) with instance state. This is no longer necessary since `Model` keeps a private `_dbe` attribute now. Can the API be changed? Should `build_reference_model` be a private function?